src/main/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategy.java

```diff
@@ -236,18 +239,20 @@ protected InetAddress mapAddress(String address) {
         }
     }
 
-    private KubernetesClient buildKubernetesClient(String token, String kubernetesMaster) {
-        if (StringUtil.isNullOrEmpty(token)) {
-            token = readAccountToken();
-        }
-        return new RetryKubernetesClient(new DefaultKubernetesClient(kubernetesMaster, token));
+    private KubernetesClient buildKubernetesClient(String kubernetesMaster, String accessToken, String caCertificate) {
+        return new RetryKubernetesClient(new DefaultKubernetesClient(kubernetesMaster, accessToken, caCertificate));
     }
 
     @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
     private static String readAccountToken() {
         return readFileContents("/var/run/secrets/kubernetes.io/serviceaccount/token");
     }
 
+    @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
+    private static String readCaCertificate() {
+        return readFileContents("/var/run/secrets/kubernetes.io/serviceaccount/ca.crt");
```

Comment:
why do we need to read from local file? What is the use case here? As far as I see it was not needed before.

Reply:
This was the same before. Reading from the local file is the scenario, when you run the client in your k8 cluster. It's here: https://github.com/hazelcast/hazelcast-kubernetes/blob/master/src/main/java/com/hazelcast/kubernetes/DefaultKubernetesClient.java#L312

Reply:
I see. Thanks!